x-pack/plugin/inference/src/main/java/org/elasticsearch/xpack/inference/chunking/SentenceBoundaryChunker.java

```diff
@@ -33,6 +36,22 @@ public SentenceBoundaryChunker() {
         wordIterator = BreakIterator.getWordInstance(Locale.ROOT);
     }
 
+    /**
+     * Break the input text into small chunks on sentence boundaries.
+     *
+     * @param input Text to chunk
+     * @param chunkingSettings Chunking settings that define maxNumberWordsPerChunk
+     * @return The input text chunked
+     */
+    @Override
+    public List<String> chunk(String input, ChunkingSettings chunkingSettings) {
+        if (chunkingSettings instanceof SentenceBoundaryChunkingSettings sentenceBoundaryChunkingSettings) {
+            return chunk(input, sentenceBoundaryChunkingSettings.maxChunkSize);
+        } else {
+            throw new IllegalArgumentException(Strings.format("SentenceBoundaryChunker can't use ChunkingSettings %s", chunkingSettings));
```

Comment:
`ChunkingSettings` doesn't have nice to string method, maybe use the strategy in the error message.

```suggestion
            throw new IllegalArgumentException(Strings.format("SentenceBoundaryChunker can't use ChunkingSettings with strategy [%s]", chunkingSettings. getChunkingStrategy()));
```